packages/taquito/src/operations/types.ts

```diff
@@ -563,6 +566,53 @@ export interface FailingNoopParams {
   basedOnBlock: BlockIdentifier;
 }
 
+export interface InternalStakingParams {
+  source: string;
+  amount: number;
+  fee?: number;
+  gasLimit?: number;
+  storageLimit?: number;
+  mutez?: boolean;
+  parameter: TransactionOperationParameter;
+}
+
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+export type StakingParams = PartialBy<InternalStakingParams, 'source'>;
+export type StakeParams = Omit<StakingParams, 'parameter'>;
+export type UnstakeParams = Omit<StakingParams, 'parameter'>;
+export type FinalizeUnstakeParams = Omit<StakingParams, 'parameter' | `amount`>;
+
+/**
+ *
+ * @throws {@link InvalidAmountError}
+ * @throws {@link InvalidAddressError}
+ * @throws {@link InvalidStakingSource}
+ */
+export const validateStakingParams = (
+  params: Omit<PartialBy<StakingParams, 'amount'>, 'parameter'>,
+  entrypoint: StakingEntrypoint
+) => {
+  if (entrypoint !== 'finalize_unstake') {
```

Comment:
iffy about his design, found it reduced readability. will we considered to explicit validate in each staking function at `packages/taquito/src/contract/rpc-contract-provider.ts`

Reply:
I think separating the validator methods is a good idea.